core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java

```diff
@@ -47,11 +47,11 @@ public class CodeGenerator {
             "META-INF/services/org.eclipse.microprofile.config.spi.ConfigSourceProvider");
 
     // used by Gradle and Maven
-    public static void initAndRun(QuarkusClassLoader classLoader,
+    public static void initAndRun(QuarkusClassLoader classLoader, Path projectDir,
             PathCollection sourceParentDirs, Path generatedSourcesDir, Path buildDir,
             Consumer<Path> sourceRegistrar, ApplicationModel appModel, Properties properties,
             String launchMode, boolean test) throws CodeGenException {
```

Comment:
@aloubyansky not sure this will help with my original issue https://github.com/quarkiverse/quarkus-cxf/issues/907 where maven is run with `-f my-module/pom.xml`

Reply:
That will be the project dir as Quarkus sees it.